src/QuadGK.jl

```diff
@@ -47,11 +47,11 @@ const rulecache = AnyDict( (Float64,7) => # precomputed in 100-bit arith.
      4.1795918367346938775510204081658e-01]) )
 
 # integration segment (a,b), estimated integral I, and estimated error E
-struct Segment
-    a::Number
-    b::Number
-    I
-    E
+struct Segment{TB<:Number,TI,TE}
+    a::TB
+    b::TB
```

Comment:
Note that this is a different constraint: currently we're saying that the types of `a` and `b` are both subtypes of `Number`, but here you're saying they have to be the _same_ subtype of `Number`. Is this intentional?

Reply:
yes indeed, I considered the boundaries of the domain would be of the same type, so this was intentional. Would there be some use cases where this is not the case?

Reply:
No, not necessarily, I just wanted to make sure the change was intentional.